cartography/intel/crxcavator/crxcavator.py

```diff
@@ -0,0 +1,220 @@
+    """
+    # the JSON returned from the CRXcavator API does not return well formatted objects
+    # instead, each object is named after it's key, making enumeration more difficult
+    # will build a cleaner object for import into graph
+
+    _data_index = 1
+    _row_index = 0
+
+    extensions = []
+    for extension in extension_json.items():
+        details = extension[_data_index][_row_index]
+        if not details:
+            continue
+        extension_id = details['extension_id']
+        version = details['version']
+        data = details['data']
+        # permissions warnings can be a list or null, if list concat to a single string
+        permission_warnings = data['webstore'].get('permission_warnings')
+        if permission_warnings:
+            permission_warnings = ', '.join(permission_warnings)
+        extensions.append({
+            'id': "{0}|{1}".format(extension_id, version),
+            'extension_id': extension_id,
+            'version': version,
+            'risk_total': data['risk'].get('total'),
+            'risk_metadata': json.dumps(data['risk'].get('metadata')),
+            'address': data['webstore'].get('address'),
+            'email': data['webstore'].get('email'),
+            'icon': data['webstore'].get('icon'),
+            'last_updated': data['webstore'].get('last_updated'),
+            'name': data['webstore'].get('name'),
+            'offered_by': data['webstore'].get('offered_by'),
+            'permissions_warnings': permission_warnings,
+            'privacy_policy': data['webstore'].get('privacy_policy'),
+            'rating': data['webstore'].get('rating'),
+            'rating_users': data['webstore'].get('rating_users'),
+            'short_description': data['webstore'].get('short_description'),
+            'size': data['webstore'].get('size'),
+            'support_site': data['webstore'].get('support_site'),
+            'users': data['webstore'].get('users'),
+            'website': data['webstore'].get('website'),
+            'type': data['webstore'].get('type'),
+            'price': data['webstore'].get('price')
+        })
+    if len(extensions) == 0:
+        raise ValueError('No extensions could be parsed from JSON data')
+    return extensions
+
+
+def load_extensions(extensions, session, update_tag):
+    """
+    Ingests the extension details into Neo4J
+    :param extensions: List of extension data to load to Neo4J
+    :param session: Neo4J session object for server communication
+    :param update_tag: Timestamp used to determine data freshness
+    :return: None
+    """
+    ingestion_cypher = """
+    UNWIND {ExtensionsData} as extension
+    MERGE (e:ChromeExtension{id: extension.id})
+    ON CREATE SET
+    e.extension_id = extension.extension_id,
+    e.version = extension.version,
+    e.firstseen = timestamp()
+    SET
+    e.extcalls = extension.extcalls,
+    e.risk_total = extension.risk_total,
```

Comment:
[schema] is there a way to represent risk as Risk node like we do for other scenarios (cve, vulndb, safedb..). Some example of question and query to identify risk would help validate the schema (made same comment for the documentation)

Reply:
Discussed in person earlier. In this context, risk score is part of the already derived data from CRXcavator and does not make sense as a separate node.

Reply:
[not pr blocker] it make sense but what about the scenario of calculating risk across asset? For example

(Risk)-[..*]->(Human) . count(risk). In the intent to measure risk, we may want to normalize. Can be done now vs later but I find it's important to think about schema and future in open source as we would introduce breaking changes others may not appreciate. CC: @ecdavis

Reply:
Agreed we should normalize at some point however at this stage I think we're fairly unclear what a good normalization would look like. I think that will become clearer as we gather data from more sources and better understand how we use that data. The Risk node as it currently exists represents a discrete issue vs. a general risk assessment. We probably want to address this discrepancy in future, but I think that's a problem for a subsequent PR.